paasta_tools/tron_tools.py

```diff
@@ -282,6 +282,32 @@ def __init__(
         self.job, self.action = decompose_instance(instance)
         # Indicate whether this config object is created for validation
         self.for_validation = for_validation
+        self.action_spark_config = None
+
+    def get_cpus(self) -> float:
+        # set Spark driver pod CPU if it is specified by Spark arguments
+        cpus = 0.0
+        if (
+            self.action_spark_config
+            and "spark.driver.cores" in self.action_spark_config
+        ):
+            cpus = float(self.action_spark_config["spark.driver.cores"])
+        # use the soa config otherwise
+        return cpus or super().get_cpus()
+
+    def get_mem(self) -> float:
+        # set Spark driver pod memory if it is specified by Spark arguments
+        mem_mb = 0.0
+        if (
+            self.action_spark_config
+            and "spark.driver.memory" in self.action_spark_config
+        ):
+            # need to set mem in MB based on tron schema
+            mem_mb = spark_tools.get_spark_memory_in_unit(
+                self.action_spark_config["spark.driver.memory"], "m"
+            )
+        # use the soa config otherwise
+        return mem_mb or super().get_mem()
```

Comment:
```suggestion
    def get_mem(self) -> float:
        if (
            self.get_executor() == "spark"
            and "spark.driver.memory" in self.action_spark_config
        ):
            return spark_tools.get_spark_memory_in_unit(
                self.action_spark_config["spark.driver.memory"], "m"
            )
        # NOTE: we fallback to this if there's no spark.driver.memory config to
        # use the paasta default
        return super().get_mem()
```

Reply:
(we probably also want to do the same thing i mentioned above re: making the schema ensure that `spark.driver.memory` is a number :))

Reply:
and i think we might need to make sure this returns an int? i don't think we allow for fractional `mem` values

Reply:
`spark.driver.memory` doesn't have to be a number. We allow values allowed by Spark, which are JVM memory strings. I saw that the signature of the original `get_mem` method returns a float so I kept the same here
Should I change it to return int at both the places??

Reply:
re: `spark.driver.memory`: i don't think that would work with the code as-is since we're hardcoding that the unit is `"m"`

that said, we can still add some schema/`paasta validate` validation so that we can ensure that we're not getting junk data from users 

re: changing the signature: we can probably leave this for another time - i see that flink is using floats for things like `mem: 1.5Gi`

Reply:
oh, re: `unit="m"` - i re-read the code with fresh eyes today and realized i'd been misreading it, the code is converting the code to mb :p

Reply:
that said, all the pods i'm spot-checking only have integer `PAASTA_RESOURCE_MEM` values and the flink pods that are setting fractional `mem` values don't add that env var - so casting to an int here is probably a good idea to keep the data looking the same (mypy shouldn't complain about returning an int from a function typed to return a float from my testing)